activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb

```diff
@@ -542,12 +542,12 @@ def type_to_sql(type, limit = nil, precision = nil, scale = nil)
               when 5..8; 'bigint'
               else raise(ActiveRecordError, "No integer type has byte size #{limit}. Use a numeric with precision 0 instead.")
             end
-          when 'datetime'
-            return super unless precision
-
+          when 'datetime', 'time'
             case precision
-              when 0..6; "timestamp(#{precision})"
-              else raise(ActiveRecordError, "No timestamp type has precision of #{precision}. The allowed range of precision is from 0 to 6")
+              when nil, 0..6; super
+              else
+                native_type = native_database_types[type.to_sym][:name]
+                raise(ActiveRecordError, "No #{native_type} type has precision of #{precision}. The allowed range of precision is from 0 to 6")
```

Comment:
Is it possible to move this behavior to the abstract adapter, since it looks to be the same?